renderer/brave_content_settings_observer_flash_browsertest.cc

```diff
@@ -0,0 +1,203 @@
+};
+
+}  // namespace
+
+class BraveContentSettingsObserverFlashBrowserTest : public InProcessBrowserTest {
+  public:
+    void SetUpOnMainThread() override {
+      InProcessBrowserTest::SetUpOnMainThread();
+
+      content_client_.reset(new ChromeContentClient);
+      content::SetContentClient(content_client_.get());
+      browser_content_client_.reset(new BraveContentBrowserClient());
+      content::SetBrowserClientForTesting(browser_content_client_.get());
+
+      host_resolver()->AddRule("*", "127.0.0.1");
+      content::SetupCrossSiteRedirector(embedded_test_server());
+
+      brave::RegisterPathProvider();
+      base::FilePath test_data_dir;
+      PathService::Get(brave::DIR_TEST_DATA, &test_data_dir);
+      embedded_test_server()->ServeFilesFromDirectory(test_data_dir);
+
+      ASSERT_TRUE(embedded_test_server()->Start());
+
+      url_ = embedded_test_server()->GetURL("a.com", "/flash.html");
+      top_level_page_pattern_ =
+          ContentSettingsPattern::FromString("http://a.com/*");
+    }
+
+      // PermissionsBrowserTest
+    void SetUpCommandLine(base::CommandLine* command_line) override {
+      InProcessBrowserTest::SetUpCommandLine(command_line);
+      ASSERT_TRUE(ppapi::RegisterFlashTestPlugin(command_line));
+      // These tests are for the permission prompt to add and remove Flash from
+      // navigator.plugins. We disable Plugin Power Saver, because its plugin
+      // throttling make it harder to test if Flash was succcessfully enabled.
+      command_line->AppendSwitchASCII(
+          switches::kOverridePluginPowerSaverForTesting, "never");
+    }
+
+    void TearDown() override {
+      browser_content_client_.reset();
+      content_client_.reset();
+    }
+
+    const GURL& url() { return url_; }
+
+    const ContentSettingsPattern& top_level_page_pattern() {
+      return top_level_page_pattern_;
+    }
+
+    const ContentSettingsPattern& empty_pattern() {
+      return empty_pattern_;
+    }
+
+    HostContentSettingsMap * content_settings() {
+      return HostContentSettingsMapFactory::GetForProfile(browser()->profile());
+    }
+
+    void UnblockFlash() {
+      content_settings()->SetContentSettingCustomScope(
+          top_level_page_pattern_,
+          ContentSettingsPattern::Wildcard(),
+          CONTENT_SETTINGS_TYPE_PLUGINS,
+          std::string(),
+          CONTENT_SETTING_DETECT_IMPORTANT_CONTENT);
+    }
+
+    void AllowFlash() {
+      content_settings()->SetContentSettingCustomScope(
+          top_level_page_pattern_,
+          ContentSettingsPattern::Wildcard(),
+          CONTENT_SETTINGS_TYPE_PLUGINS,
+          std::string(),
+          CONTENT_SETTING_ALLOW);
+    }
+
+    content::WebContents* contents() {
+      return browser()->tab_strip_model()->GetActiveWebContents();
+    }
+
+    bool RunScriptReturnBool(const std::string &script) {
+      bool value;
+      EXPECT_TRUE(ExecuteScriptAndExtractBool(contents(), script, &value));
```

Comment:
nit: generally I would prefer to avoid putting EXPECT/ASSERT codes inside a function since it would be harder to see where it actually failed when you call it multiple times in a single test. I'm OK with it since currently it is called only once.

Reply:
it's only used once anyway, I'll just not have that helper.